### PR TITLE
Update to 24.08.0

### PR DESCRIPTION
**Pull request by attah**
Now somewhat presentable... but some conflict prevents me from testing.

### CODE REVIEW

File: rpm/0003-Revert-Async-api-for-certificate-validation.patch [hunk 1]
```diff
@@ -0,0 +1,1171 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Anton Thomasson <antonthomasson@gmail.com>
+Date: Sat, 29 Jun 2024 22:09:09 +0200
+Subject: [PATCH] Revert "Async api for certificate validation"
```
Comment (svuorela) [inline rpm/0003-Revert-Async-api-for-certificate-validation.patch]: As the one who did this upstream, given Sailfish' requirements I think it is an okay solution to revert this.

Also, internally, the async certificate validation is only fully implemented in the in-sailfish disabled gpgme backend.

It is also only newest Okular that so far uses it, and that Okular is also based on Qt6.

**Comment by attah** [comment]
I'll look at bumping this to 24.08 when it rolls around in probably 1-2 weeks.
There was some mistake with the so file version in 06 that i missed, and some new fixes that looks not bad to have.
Though already as-is it fixes CVE-2024-6239.

File: rpm/poppler.spec [hunk 2]
```diff
@@ -101,7 +103,7 @@ and pdffonts (PDF font analyzer).
 %autosetup -p1 -n %{name}-%{version}/upstream
 
 %build
-%cmake . -G Ninja -DENABLE_BOOST=ON
+%cmake . -G Ninja -DENABLE_BOOST=ON -DENABLE_GPGME=off -DENABLE_QT6=off -DENABLE_LCMS=off -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations"
```
Comment (pvuorela) [inline rpm/poppler.spec]: For small detail, why the -Wno-deprecated-declarations? I'm not seeing much warnings without it either.

File: rpm/poppler.spec [hunk 3]
```diff
@@ -156,3 +158,7 @@ rm -f %{buildroot}%{_libdir}/*.la
 %defattr(-,root,root,-)
 %{_bindir}/pdf*
 %{_mandir}/man1/*
+
+%exclude
+%{_datarootdir}/locale/ca/LC_MESSAGES/pdfsig.mo
```
Comment (pvuorela) [inline rpm/poppler.spec]: Commonly these are in one line like "%exclude /foo/bar". 

Interesting thing this one separate catalan translation in the whole repository.

File: rpm/poppler.spec [hunk 4]
```diff
@@ -29,6 +29,8 @@ BuildRequires:  boost-devel >= 1.71.0
 
 Patch1: 0001-Revert-Require-the-newer-qt5-provided-by-the-new-bas.patch
 Patch2: 0002-Don-t-build-qt5-demos-tests.patch
+Patch3: 0003-Revert-Async-api-for-certificate-validation.patch
+Patch4: 0004-Use-old-way-to-get-seconds-since-epoch.patch
```
Comment (pvuorela) [inline rpm/poppler.spec]: Could consider merging this to the first patch, it's more or less about the same thing. Either way.

**Review by pvuorela — COMMENTED** [review]
Seems mostly fine, just a few small remarks. Could even approve already, but I'll wait a moment if you want to address some.

File: rpm/poppler.spec (hunk 2 above)
Comment (attah) [reply rpm/poppler.spec]: There are only 10 or so... but somehow when i did the uplift the first time around they were fatal. Or maybe i was confused and managed to hide the true issue long enough to think this was a working fix.

File: rpm/poppler.spec (hunk 3 above)
Comment (attah) [reply rpm/poppler.spec]: The maintainers are really in to their translations, and probably Catalan.

File: rpm/poppler.spec (hunk 4 above)
Comment (attah) [reply rpm/poppler.spec]: Reshuffled and merged.

File: rpm/poppler.spec (hunk 3 above)
Comment (attah) [reply rpm/poppler.spec]: onelined as well

**Review by mlehtima — APPROVED** [review]

**Review by pvuorela — APPROVED** [review]

**Pull request merged**